directory/directory_test.go

```diff
@@ -0,0 +1,71 @@
+package directory
+
+import (
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCanonicalDockerReference(t *testing.T) {
+	dest := NewDirImageDestination("/path/to/somewhere")
+	_, err := dest.CanonicalDockerReference()
+	assert.NotNil(t, err)
+}
+
+func TestGetPutManifest(t *testing.T) {
+	tmp, err := ioutil.TempDir("", "put-manifest")
+	assert.Nil(t, err)
+	dest := NewDirImageDestination(tmp)
+	err = dest.PutManifest([]byte("test-manifest"))
+	assert.Nil(t, err)
+	src := NewDirImageSource(tmp)
+	m, mt, err := src.GetManifest(nil)
+	assert.Nil(t, err)
+	assert.Equal(t, []byte("test-manifest"), m)
+	assert.Equal(t, "", mt)
+}
+
+func TestGetPutBlob(t *testing.T) {
+	tmp, err := ioutil.TempDir("", "put-blob")
+	assert.Nil(t, err)
+	dest := NewDirImageDestination(tmp)
+	err = dest.PutBlob("digest-test", strings.NewReader("test-blob"))
+	assert.Nil(t, err)
+	src := NewDirImageSource(tmp)
+	rc, size, err := src.GetBlob("digest-test")
+	assert.Nil(t, err)
+	b, err := ioutil.ReadAll(rc)
+	assert.Nil(t, err)
+	assert.Equal(t, []byte("test-blob"), b)
+	assert.Equal(t, int64(len("test-blob")), size)
+}
+
+func TestGetPutSignatures(t *testing.T) {
+	tmp, err := ioutil.TempDir("", "put-signatures")
+	assert.Nil(t, err)
+	dest := NewDirImageDestination(tmp)
+	signatures := [][]byte{}
+	signatures = append(signatures, []byte("sig1"))
+	signatures = append(signatures, []byte("sig2"))
+	err = dest.PutSignatures(signatures)
+	assert.Nil(t, err)
+	src := NewDirImageSource(tmp)
+	sigs, err := src.GetSignatures()
+	assert.Nil(t, err)
+	assert.Equal(t, signatures, sigs)
+}
+
+func TestDelete(t *testing.T) {
+	src := NewDirImageSource("/somewhere")
+	err := src.Delete()
+	assert.NotNil(t, err)
```

Comment:
Better to use a temporary directory as well, what if someone implemented this method and the test started deleting user data?